core/src/main/java/io/ap4k/config/ConfigKey.java

```diff
@@ -44,4 +46,19 @@ public String getName() {
   public T getDefaultValue() {
     return defaultValue;
   }
+
```

Comment:
Why do we need `equals(Object o) and hasCode()` ?

Reply:
In order to exclude `defaultValue, which can't participate in the key.